src/components/settings/NumberSetting.tsx

```diff
@@ -98,6 +98,17 @@ export const NumberSetting = ({
         [value, handleUpdate, closeDialog],
     );
 
+    const submit = useCallback(
+        (newValue: number) => {
+            handleUpdate(newValue);
+            if (handleSubmit) {
+                handleSubmit(newValue);
+            }
+            closeDialog(false);
+        },
+        [value, handleUpdate, handleSubmit, closeDialog],
+    );
+
     useEffect(() => {
```

Comment:
this causes to many state updates during the live update

tbh, don't remember why I added this (I think to update the dialog value in case there was an update to it while the dialog was open?), probably can be completely removed

Reply:
what about this?

Reply:
Tbh I don't really understand `useEffect`. So just remove it?

Reply:
yeah, if everything works, just remove it

useEffect gets called after every render
here it would reset the current tmp dialog value to the original value in case it has changed